coinlore/src/adapter.ts

```diff
@@ -0,0 +1,40 @@
+import { Requester, Validator, AdapterError } from '@chainlink/external-adapter'
+import { Config, ExecuteWithConfig, ExecuteFactory } from '@chainlink/types'
+import { makeConfig, DEFAULT_ENDPOINT } from './config'
+import { global } from './endpoint'
+
+const inputParams = {
+  endpoint: false,
+}
+
+export const execute: ExecuteWithConfig<Config> = async (request, config) => {
+  const validator = new Validator(request, inputParams)
+  if (validator.error) throw validator.error
+
+  Requester.logConfig(config)
+
+  const jobRunID = validator.validated.id
+  const endpoint = validator.validated.data.endpoint || DEFAULT_ENDPOINT
+
+  switch (endpoint.toLowerCase()) {
+    case 'dominance':
+    case global.NAME: {
+      return await global.execute(request, config)
+    }
+    case 'globalmarketcap': {
+      request.data.field = 'total_mcap'
+      return await global.execute(request, config)
+    }
```

Comment:
Looks like there could be something wrong here, does the 'dominance' case do anything?